src/lib.rs

```diff
@@ -176,14 +177,22 @@ pub use poll::Poll;
 // Primitive futures
 mod done;
 mod empty;
+#[cfg(not(feature = "never"))]
 mod failed;
+#[cfg(not(feature = "never"))]
```

Comment:
Yes, I definitely agree it's best to keep this as a non-breaking change. What if we just named them something temporary like `AlwaysFinished` and `AlwaysFailed`? It's not very ergonomic, but this should just be temporary pending stabilization.